sdk/src/test/java/com/google/cloud/dataflow/sdk/runners/TemplatingDataflowPipelineRunnerTest.java

```diff
@@ -0,0 +1,123 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.options.DataflowPipelineOptions;
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import com.google.cloud.dataflow.sdk.options.PipelineOptionsFactory;
+import com.google.cloud.dataflow.sdk.testing.ExpectedLogs;
+import com.google.cloud.dataflow.sdk.testing.TestDataflowPipelineOptions;
+import com.google.cloud.dataflow.sdk.util.NoopPathValidator;
+import com.google.cloud.dataflow.sdk.util.TestCredential;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.io.File;
+
+/**
+ * Tests for TemplatingDataflowPipelineRunner.
+ */
+@RunWith(JUnit4.class)
+public class TemplatingDataflowPipelineRunnerTest {
+
+  @Mock private DataflowPipelineJob mockJob;
+
+  @Rule
+  public ExpectedLogs expectedLogs = ExpectedLogs.none(TemplatingDataflowPipelineRunner.class);
+
+  @Rule
+  public ExpectedException expectedThrown = ExpectedException.none();
+
+  @Rule
+  public final TemporaryFolder tmpDir = new TemporaryFolder();
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+    when(mockJob.getProjectId()).thenReturn("project");
+    when(mockJob.getJobId()).thenReturn("job");
+  }
+
+  /**
+   * Returns a {@link TemplatingDataflowPipelineRunner} that will return the provided a job to
+   * return. Some {@link PipelineOptions} will be extracted from the job, such as the project ID.
+   */
+  private TemplatingDataflowPipelineRunner createMockRunner(
+      String filePath) throws Exception {
+    DataflowPipelineRunner mockRunner = mock(DataflowPipelineRunner.class);
+    TestDataflowPipelineOptions options =
+        PipelineOptionsFactory.as(TestDataflowPipelineOptions.class);
+    options.setProject(mockJob.getProjectId());
+    options.setDataflowJobFile(filePath);
+    when(mockRunner.run(isA(Pipeline.class))).thenReturn(mockJob);
+
+    return new TemplatingDataflowPipelineRunner(mockRunner, options);
+  }
+
+  /**
+   * Tests that the {@link TemplatingDataflowPipelineRunner} returns normally when a template is
+   * successfully written.
+   */
+  @Test
+  public void testLoggedCompletion() throws Exception {
+    File existingFile = tmpDir.newFile();
+    createMockRunner(existingFile.getPath()).run(DirectPipeline.createForTest());
+    expectedLogs.verifyInfo("Template successfully created");
+  }
+
+  /**
+   * Tests that the {@link TemplatingDataflowPipelineRunner} throws the appropriate exception when
+   * an output file is not writable.
+   */
+  @Test
+  public void testLoggedErrorForFile() throws Exception {
+    expectedThrown.expect(RuntimeException.class);
```

Comment:
suggest moving the `expectedThrown.expect` down to right before the call that should throw. Especially for errors like RuntimeException, it is also helpful to add an expected message. Otherwise, if `options.setJobName(...)` decided to throw a RuntimeException this test would still pass.